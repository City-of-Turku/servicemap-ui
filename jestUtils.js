import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });