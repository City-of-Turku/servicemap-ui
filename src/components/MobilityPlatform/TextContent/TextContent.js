import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../styled/styled';

const TextContent = ({ titleId, translationId }) => {
  const intl = useIntl();

  const singleValTypo = (messageId, isTitle) => (
    <Typography variant={isTitle ? 'subtitle1' : 'body2'} component={isTitle ? 'h3' : 'p'}>
      {intl.formatMessage({
        id: messageId,
      })}
    </Typography>
  );

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        {singleValTypo(titleId, true)}
      </StyledHeaderContainer>
      <StyledTextContainer>
        {singleValTypo(translationId, false)}
      </StyledTextContainer>
    </StyledContainer>
  );
};

TextContent.propTypes = {
  titleId: PropTypes.string,
  translationId: PropTypes.string,
};

TextContent.defaultProps = {
  titleId: '',
  translationId: '',
};

export default TextContent;
