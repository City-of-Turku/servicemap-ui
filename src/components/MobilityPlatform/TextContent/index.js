import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import styles from './styles';
import TextContent from './TextContent';

export default withStyles(styles)(injectIntl(TextContent));
