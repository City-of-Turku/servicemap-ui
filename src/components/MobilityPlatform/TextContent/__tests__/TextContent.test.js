// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../jestUtils';
import finnishTranslations from '../../../../i18n/fi';
import TextContent from '../index';

const mockProps = {
  titleId: 'mobilityPlatform.content.scooters.noParkingAreas.title',
  translationId: 'mobilityPlatform.info.scooters.noParking',
};

const renderWithProviders = getRenderWithProviders({});

describe('<TextContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<TextContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<TextContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    const h6 = container.querySelector('h6');
    expect(h6.textContent).toContain(finnishTranslations['mobilityPlatform.content.scooters.noParkingAreas.title']);
    expect(p[0].textContent).toContain(finnishTranslations['mobilityPlatform.info.scooters.noParking']);
  });
});
