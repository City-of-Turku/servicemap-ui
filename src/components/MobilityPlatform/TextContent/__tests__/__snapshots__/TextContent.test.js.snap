// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextContent /> should work 1`] = `
<div>
  <div
    class="injectIntl(TextContent)-container-1"
  >
    <div
      class="injectIntl(TextContent)-headerContainer-2"
    >
      <div>
        <h3
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Pysäköintikieltoalue
        </h3>
      </div>
    </div>
    <div
      class="injectIntl(TextContent)-textContainer-3"
    >
      <div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Sähköpotkulautojen pysäköinti kartalla näkyville alueille on kielletty.
        </p>
      </div>
    </div>
  </div>
</div>
`;
